lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -837,6 +898,22 @@ impl Instance {
         });
 
         self.with_signals_on(|i| {
+            // set up the guest to set itself as terminable, then continue to
+            // whatever guest code we want to run
+            //
+            // lucet_context_activate takes the guest code address in `rsi`,
+            // replacing the guest return address with itself so it can run
```

Comment:
replacing the guest return address where? and `itself` is unclear here
...
ah, the comments below make it more clear, but maybe this could be reworded still?

Reply:
rephrased this comment, I think it's better but I'll leave that up to you